src/Engine/Component/CameraComponents/CameraComponent.cpp

```diff
@@ -66,81 +73,82 @@ void CameraComponent::update(float dt) {
     rightPlanePoint = nearPlanePoint + u * nearPlaneWidth / 2.f;
     rightPlaneNormal = glm::normalize(glm::cross(v,
         rightPlanePoint - goPos));
-
-    if (isDirty) {
-        this->projection = glm::perspective(fov, Window::getAspectRatio(), near, far);
-        this->view = glm::lookAt(gameObject->getSpatial()->position(), lookAt, glm::vec3(0, 1, 0));
-        isDirty = false;
-    }
 }
 
-const bool CameraComponent::sphereInFrustum(glm::vec3 center, float radius) const {
+const bool CameraComponent::sphereInFrustum(const Sphere & sphere) const {
```

Comment:
Can we move this function out of CameraComponent? CameraComponent should solely manage the Camera's data and transformations - it shouldn't be responsible for any actual frustum culling.
This should be fine for now if we can't move it, but as the RenderSystem gets up to speed with frustum culling (my changes are incoming) then I think a lot of this functionality should move there as well.

Reply:
i agree. I was mostly just amending some of Brenan's code without overhauling it

Reply:
@barlebino

Reply:
sorry for late response. yes we can move this function elsewhere. will we move the frustum data as well?

Reply:
I think frustum data can stay with the camera - camera owns the projection and view matrix as well. The actual computation and culling should move to render system.